### PR TITLE
feat: add service_account_token resource

### PR DESCRIPTION
**Pull request by srevinsaju**
* adds a new resource `doppler_service_account_token`
* added example usage
* autogenerated docs for newly added resource

### Sample deployment 
```tf
terraform {
  required_providers {
    doppler = {
      source = "doppler.com/core/doppler"
      version = "0.0.1"
    }
  }
}

provider "doppler" {
  doppler_token = "..."
}

resource "doppler_service_account" "ci" {
  name = "ci"
  workplace_role = "no_access"
}

resource "doppler_service_account_token" "builder_ci_token" {
  service_account_slug = doppler_service_account.ci.slug
  name            = "Builder CI Token"
  expires_at      = "2024-05-30T11:00:00.000Z"
}

output "service_account_token_slug" {
  value = doppler_service_account_token.builder_ci_token.slug
}

output "service_account_token_api_key" {
  sensitive = true
  value = doppler_service_account_token.builder_ci_token.api_key
}
```

What is tested:
* Creation (with expiry date)
* Creation (without expiry date)
* Update from no expiry date to having an expiry date, and vice versa - which should do a full CRUD test

### CODE REVIEW

File: doppler/api.go [hunk 1]
```diff
@@ -835,6 +835,50 @@ func (client APIClient) DeleteServiceAccount(ctx context.Context, slug string) e
 	return nil
 }
 
+// Service Account Tokens
+
+func (client APIClient) GetServiceAccountToken(ctx context.Context, serviceAccountSlug string, slug string) (ServiceAccountTokenResponse, error) {
+	response, err := client.PerformRequestWithRetry(ctx, "GET", fmt.Sprintf("/v3/workplace/service_accounts/service_account/%s/tokens/token/%s", serviceAccountSlug, slug), []QueryParam{}, nil)
```
Comment (nmanoogian) [inline doppler/api.go:841]: For good measure, can we escape all URL params?

```diff
diff --git a/doppler/api.go b/doppler/api.go
index 9fb6594..06daeb5 100644
--- a/doppler/api.go
+++ b/doppler/api.go
@@ -838,7 +838,7 @@ func (client APIClient) DeleteServiceAccount(ctx context.Context, slug string) e
 // Service Account Tokens
 
 func (client APIClient) GetServiceAccountToken(ctx context.Context, serviceAccountSlug string, slug string) (ServiceAccountTokenResponse, error) {
-	response, err := client.PerformRequestWithRetry(ctx, "GET", fmt.Sprintf("/v3/workplace/service_accounts/service_account/%s/tokens/token/%s", serviceAccountSlug, slug), []QueryParam{}, nil)
+	response, err := client.PerformRequestWithRetry(ctx, "GET", fmt.Sprintf("/v3/workplace/service_accounts/service_account/%s/tokens/token/%s", url.QueryEscape(serviceAccountSlug), url.QueryEscape(slug)), []QueryParam{}, nil)
 	if err != nil {
 		return ServiceAccountTokenResponse{}, err
 	}
@@ -860,7 +860,7 @@ func (client APIClient) CreateServiceAccountToken(ctx context.Context, serviceAc
 	if err != nil {
 		return nil, &APIError{Err: err, Message: "Unable to serialize service token"}
 	}
-	response, err := client.PerformRequestWithRetry(ctx, "POST", fmt.Sprintf("/v3/workplace/service_accounts/service_account/%s/tokens", serviceAccountSlug), []QueryParam{}, body)
+	response, err := client.PerformRequestWithRetry(ctx, "POST", fmt.Sprintf("/v3/workplace/service_accounts/service_account/%s/tokens", url.QueryEscape(serviceAccountSlug)), []QueryParam{}, body)
 	if err != nil {
 		return nil, err
 	}
@@ -872,7 +872,7 @@ func (client APIClient) CreateServiceAccountToken(ctx context.Context, serviceAc
 }
 
 func (client APIClient) DeleteServiceAccountToken(ctx context.Context, serviceAccountSlug string, slug string) error {
-	_, err := client.PerformRequestWithRetry(ctx, "DELETE", fmt.Sprintf("/v3/workplace/service_accounts/service_account/%s/tokens/token/%s", serviceAccountSlug, slug), []QueryParam{}, nil)
+	_, err := client.PerformRequestWithRetry(ctx, "DELETE", fmt.Sprintf("/v3/workplace/service_accounts/service_account/%s/tokens/token/%s", url.QueryEscape(serviceAccountSlug), url.QueryEscape(slug)), []QueryParam{}, nil)
 	if err != nil {
 		return err
 	}


```

File: doppler/resource_service_account_token.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	var diags diag.Diagnostics
+	serviceAccount := d.Get("service_account_slug").(string)
+	slug := d.Id()
+
+	token, err := client.GetServiceAccountToken(ctx, serviceAccount, slug)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err = d.Set("expires_at", token.ServiceAccountToken.ExpiresAt); err != nil {
```
Comment (nmanoogian) [inline doppler/resource_service_account_token.go:103]: Even though the name cannot be changed, can we fetch it here to ensure it hasn't been changed externally?

```diff
diff --git a/doppler/resource_service_account_token.go b/doppler/resource_service_account_token.go
index e0b55f9..c05a9d6 100644
--- a/doppler/resource_service_account_token.go
+++ b/doppler/resource_service_account_token.go
@@ -100,6 +100,9 @@ func resourceServiceAccountTokenRead(ctx context.Context, d *schema.ResourceData
 		return diag.FromErr(err)
 	}
 
+	if err = d.Set("name", token.ServiceAccountToken.Name); err != nil {
+		return diag.FromErr(err)
+	}
 	if err = d.Set("expires_at", token.ServiceAccountToken.ExpiresAt); err != nil {
 		return diag.FromErr(err)
 	}


```

File: go.mod [hunk 3]
```diff
@@ -5,40 +5,74 @@ go 1.21
 require github.com/hashicorp/terraform-plugin-sdk/v2 v2.33.0
 
 require (
+	github.com/BurntSushi/toml v1.2.1 // indirect
```
Comment (nmanoogian) [inline go.mod:8]: Can we revert the `go.mod` and `go.sum` changes unless necessary?

**Comment by srevinsaju** [comment]
Thanks for the review @nmanoogian, I have updated with your suggested patches!

**Review by nmanoogian — APPROVED** [review]

**Pull request merged**